### PR TITLE
[CARBONDATA-3033][Doc]Fix bugs in documents

### PR DESCRIPTION
**Pull request by xuchuanyin**
1. add support for s3 as carbon.storelocation
2. optimize the description for unsafe memory
3. fix errors in description for 'carbon.sort.file.write.buffer.size'
4. fix the incorrect default value for 'carbon.csv.read.buffersize.byte' in code
5. reorder and group the parameters for data loading by its function scope

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [x] Any interfaces changed?
 `NO`
 - [x] Any backward compatibility impacted?
  `NO`
 - [x] Document update required?
 `NO`
 - [x] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
 `NO`
        - How it is tested? Please attach test report.
 `NO`
        - Is it a performance related change? Please attach the performance test report.
 `NO`
        - Any additional information to help reviewers in testing this change.
        `NO`
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
 `NA`

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/921/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/1130/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/9191/

File: core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java [hunk 1]
```diff
@@ -361,7 +361,7 @@
   /**
    * CSV_READ_BUFFER_SIZE
    */
-  public static final String CSV_READ_BUFFER_SIZE_DEFAULT = "50000";
+  public static final String CSV_READ_BUFFER_SIZE_DEFAULT = "1048576";
```
Comment (QiangCai) [inline core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java]:
```suggestion
  public static final String CSV_READ_BUFFER_SIZE_DEFAULT = "1048576"; // 1mb
```

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/950/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/9215/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/1158/

**Pull request closed without merging**